src/Umbraco.Web.BackOffice/Controllers/MediaController.cs

```diff
@@ -768,11 +780,51 @@ public async Task<IActionResult> PostAddFile([FromForm] string path, [FromForm]
                             _mediaService.Save(folderMediaItem);
                         }
                     }
+
                     //set the media root to the folder id so uploaded files will end there.
                     parentId = folderMediaItem.Id;
                 }
             }
 
+            var mediaTypeAlias = string.Empty;
+            var allMediaTypes = _mediaTypeService.GetAll().ToList();
+            var allowedContentTypes = new HashSet<IMediaType>();
+
+            if (parentId != Constants.System.Root)
+            {
+                var mediaFolderItem = _mediaService.GetById(parentId.Value);
+                var mediaFolderType =
+                    allMediaTypes.FirstOrDefault(x => x.Alias == mediaFolderItem.ContentType.Alias);
+
+                if (mediaFolderType != null)
+                {
+                    IMediaType mediaTypeItem = null;
+
+                    foreach (ContentTypeSort allowedContentType in mediaFolderType.AllowedContentTypes)
+                    {
+                        IMediaType checkMediaTypeItem = _mediaTypeService.Get(allowedContentType.Id.Value);
```

Comment:
Could reusue allMediaTypes here and save a call to the mediatype service

Reply:
This method is very long and will be hard to maintain. Could this part that is responsible for generating the allowed content types be made into it's own method?